app/src/main/java/org/dpppt/android/app/network/CheckConnectionStatus.java

```diff
@@ -0,0 +1,103 @@
+package org.dpppt.android.app.network;
+
+import android.app.job.JobParameters;
+import android.app.job.JobService;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.os.Build;
+
+import org.dpppt.android.app.util.DebugUtils;
+
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import org.dpppt.android.app.R;
+import org.dpppt.android.app.notifications.NotificationService;
+
+
+public class CheckConnectionStatus extends JobService {
```

Comment:
This class set the listener for network status, try until the connection return alive atleast one time